ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadWork.java

```diff
@@ -65,26 +67,36 @@ public ReplLoadWork(HiveConf hiveConf, String dumpDirectory, String dbNameToLoad
     this.dbNameToLoadIn = dbNameToLoadIn;
     rootTask = null;
     if (isIncrementalDump) {
-      incrementalIterator = new IncrementalLoadEventsIterator(dumpDirectory, hiveConf);
-      this.bootstrapIterator = null;
-      this.constraintsIterator = null;
-      incrementalLoad =
+      incrementalLoadTasksBuilder =
           new IncrementalLoadTasksBuilder(dbNameToLoadIn, tableNameToLoadIn, dumpDirectory,
-              incrementalIterator, hiveConf, eventTo);
+                  new IncrementalLoadEventsIterator(dumpDirectory, hiveConf), hiveConf, eventTo);
+
+      /*
+       * If the current incremental dump also includes bootstrap for some tables, then create iterator
+       * for the same.
+       */
+      Path incBootstrapDir = new Path(dumpDirectory, ReplUtils.INC_BOOTSTRAP_ROOT_DIR_NAME);
+      FileSystem fs = incBootstrapDir.getFileSystem(hiveConf);
+      if (fs.exists(incBootstrapDir)) {
+        this.bootstrapIterator = new BootstrapEventsIterator(incBootstrapDir.toString(), dbNameToLoadIn, hiveConf);
+        this.constraintsIterator = new ConstraintEventsIterator(dumpDirectory, hiveConf);
```

Comment:
executeBootstrapLoad method expects it to be not null even if constraints are not there. Also, I set it here so that in future if bootstrapped table (such as ACID tables) have constraints, it works fine.